core/src/main/java/ai/timefold/solver/core/impl/localsearch/decider/acceptor/restart/AbstractGeometricStuckCriterion.java

```diff
@@ -18,29 +18,28 @@
  * 
  * @param <Solution_> the solution type
  */
-public abstract class AbstractGeometricRestartStrategy<Solution_> implements RestartStrategy<Solution_> {
+public abstract class AbstractGeometricStuckCriterion<Solution_> implements StuckCriterion<Solution_> {
+    protected static final Logger logger = LoggerFactory.getLogger(AbstractGeometricStuckCriterion.class);
     private static final double GEOMETRIC_FACTOR = 1.4; // Value extracted from the cited paper
-    protected final Clock clock;
-    protected final Logger logger = LoggerFactory.getLogger(AbstractGeometricRestartStrategy.class);
-    protected final double scalingFactor;
+    private static final long GRACE_PERIOD_MILLIS = 30_000; // Same value as DiminishedReturnsTermination
 
+    private final double scalingFactor;
+    private final Clock clock;
     private boolean gracePeriodFinished;
-    private boolean restartTriggered;
     private long gracePeriodMillis;
     protected long nextRestart;
-    protected double currentGeometricGrowFactor;
+    private double currentGeometricGrowFactor;
 
-    protected AbstractGeometricRestartStrategy(Clock clock, double scalingFactor) {
+    protected AbstractGeometricStuckCriterion(Clock clock, double scalingFactor) {
         this.clock = clock;
         this.scalingFactor = scalingFactor;
     }
 
     @Override
     public void phaseStarted(LocalSearchPhaseScope<Solution_> phaseScope) {
-        restartTriggered = false;
         if (gracePeriodMillis == 0) {
-            // 10 seconds of grace period
-            gracePeriodMillis = clock.millis() + 10_000;
+            // 30 seconds of grace period
+            gracePeriodMillis = clock.instant().toEpochMilli() + GRACE_PERIOD_MILLIS;
```

Comment:
My earlier comment to not use `clock.millis()` was intending to suggest that `gracePeriod` should be an `Instant` (so something like `gracePeriodEnd`), and you can use `Instant.plusMillis(...)` to modify it.

`Instant` has semantics, `long` does not.